reactor-netty-http/src/main/java/reactor/netty/http/client/Http2Pool.java

```diff
@@ -624,13 +624,23 @@ void pendingAcquireLimitReached(Borrower borrower, int maxPending) {
 	 * @param borrower a new {@link Borrower} to add to the queue and later either serve or consider pending
 	 */
 	void pendingOffer(Borrower borrower) {
-		int maxPending = poolConfig.maxPending();
 		ConcurrentLinkedDeque<Borrower> pendingQueue = pending;
 		if (pendingQueue == TERMINATED) {
 			return;
 		}
+
 		int postOffer = addPending(pendingQueue, borrower, false);
 
+		long estimateStreamsCount = totalMaxConcurrentStreams - acquired;
+		int permits = poolConfig.allocationStrategy().estimatePermitCount();
+		if (permits + estimateStreamsCount < postOffer) {
+			borrower.pendingAcquireStart = clock.millis();
+			if (!borrower.acquireTimeout.isZero()) {
+				borrower.timeoutTask = poolConfig.pendingAcquireTimer().apply(borrower, borrower.acquireTimeout);
+			}
+		}
+
+		int maxPending = poolConfig.maxPending();
```

Comment:
overall, it seems good to me.

I just wonder why the `maxPending` variable has been moved to line 643, it's probably for readability reasons.
then since the `maxPending` variable is only used if `WIP.getAndIncrement(this) == 0`, then why not declaring this variable in the if block ? or am I missing the point ? something like:

```
		if (WIP.getAndIncrement(this) == 0) {
			int maxPending = poolConfig.maxPending();
            ...
```

Reply:
I moved it to the `if`, thanks